### PR TITLE
[p2p] add godoc comments to Listener

### PR DESCRIPTION
**Pull request by melekes**

### CODE REVIEW

**Comment by xla** [comment]
@melekes Two nodes have been connected, but what about implications on addrbook and pex?

**Comment by melekes** [comment]
> what about implications on addrbook and pex?

should be good. addrbook is using `NetAddress.String()`, which calls `IP.String()`. maybe we can write a test though

File: p2p/listener.go [hunk 1]
```diff
@@ -147,21 +159,26 @@ func (l *DefaultListener) listenRoutine() {
...
 }
 
+// ExternalAddress returns the external NetAddress (publicly available,
+// determined using either UPnP or local resolver).
 func (l *DefaultListener) ExternalAddress() *NetAddress {
 	return l.extAddr
 }
 
-func (l *DefaultListener) ExternalAddressToString() string {
+// TODO: remove the need for this function by supporting IPv6 in NetAddress.
```
Comment (melekes) [inline p2p/listener.go]: actually, we may already support IPv6 in NetAddress. `net.IP` can hold either IPv4 or IPv6.

**Comment by melekes** [comment]
@xla what's exactly you don't like in this PR? changelog entry? I can remove it

**Comment by xla** [comment]
@melekes If all components are supporting ipv6 then I all my concerns are adressed.

**Comment by xla** [comment]
Sorry bad wording, what I meant is that this should be fine as nothing seem to break from it, so good to goz

**Pull request merged**